### ISSUE TITLE
Clean up branches on this repo

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#17716]
There are a lot of extra branches on this repo. We should clean them up. Pull requests should be done from forks, not from branches on this repo. Unfortunately, in some cases, the branches are created automatically by GitHub, e.g., when you revert a PR. If a branch has an open PR, it won't be possible to switch it to a fork without opening a new PR. If the PR is closed, the branch should be deleted.

**Comment by sylee957** [comment]
>  Unfortunately, in some cases, the branches are created automatically by GitHub, e.g., when you revert a PR. If a branch has an open PR, it won't be possible to switch it to a fork without opening a new PR.

Do you mean that `revert-14280-zero_vector` and `revert-15373-matexpr_expand` are created by this reason? Well, then those branches can be left behind.
But we may have to contact @smichr to close the other stale branches.

**Comment by asmeurer** [comment]
Yeah the `revert` branches are created automatically by GitHub whenever you click the "revert pull request" button.
